dao/elasticsearch/service.go

```diff
@@ -25,14 +24,21 @@ import (
 )
 
 const (
-	userLockTimeout        = time.Second
-	userLockTimeoutMessage = "Cannot interrupt system operation (%s). Try again later."
+	userLockTimeout = time.Second
 )
 
+type ErrSystemBusy struct {
+	blocker string
+}
+
+func (e ErrSystemBusy) Error() string {
+	return fmt.Sprintf("Cannot interrupt system operation (%s). Try again later.", e.blocker)
+}
+
 // AddService adds a new service. Returns an error if service already exists.
 func (this *ControlPlaneDao) AddService(svc service.Service, serviceId *string) error {
 	if gotLock, blocker := this.facade.DFSLock(datastore.Get()).LockWithTimeout("add service", userLockTimeout); !gotLock {
```

Comment:
I wanted TimedMutex to be more generic and usable in multiple situations. The 'holder' field does not have to be a human-readable string for use in error messages--I'm just using it that way here. (And that's why the errors are defined at the application level, not within the mutex.)